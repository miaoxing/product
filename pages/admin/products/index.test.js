import Index from './index';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router';
import $ from 'miaoxing';
import {bootstrap, createPromise, setUrl, resetUrl} from '@mxjs/test';
import {app} from '@mxjs/app';

bootstrap();

describe('admin/products', () => {
  beforeEach(function () {
    setUrl('admin/products');
    app.page = {
      collection: 'admin/products',
      index: true,
    };
  });

  afterEach(() => {
    resetUrl();
    app.page = {};
  });

  test('index', async () => {
    const promise = createPromise();

    $.http = jest.fn()
      // 读取列表数据
      .mockImplementationOnce(() => promise.resolve({
        ret: {
          code: 1,
          data: [
            {
              id: 1,
              name: '商品1',
              minPrice: 3,
              stockNum: 4,
              isListing: true,
              createdAt: '2020-01-01 00:00:00',
            },
          ],
        },
      }));

    const {findByText} = render(<MemoryRouter>
      <Index/>
    </MemoryRouter>);

    await findByText('商品1');
    await findByText('￥3');

    await Promise.all([promise]);
    expect($.http).toHaveBeenCalledTimes(1);
    expect($.http).toMatchSnapshot();
  });
});
