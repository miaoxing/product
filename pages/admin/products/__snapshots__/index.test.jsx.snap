// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin/products index 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "method": "get",
        "url": "products?page=1&limit=20",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
  ],
}
`;
